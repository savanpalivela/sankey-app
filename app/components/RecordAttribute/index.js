import RecordAttribute from './recordAttribute.component';

export default RecordAttribute;
