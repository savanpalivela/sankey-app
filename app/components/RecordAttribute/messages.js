/*
 * RecordAttribute Messages
 *
 * This contains all the text for the RecordAttribute component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.RecordAttribute';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the RecordAttribute component!',
  },
});
