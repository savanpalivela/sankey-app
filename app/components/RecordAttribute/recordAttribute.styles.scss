.attibuteItem {
  margin: 10px 0;

  .recordKeyText {
    padding: 6px 35px;
  }
}