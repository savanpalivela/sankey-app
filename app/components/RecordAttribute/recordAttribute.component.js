/**
 *
 * RecordAttribute
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Trash } from 'react-bootstrap-icons';

import { FormattedMessage } from 'react-intl';
import messages from './recordAttribute.messages';
import styles from './recordAttribute.styles.scss';

function RecordAttribute({
  keyText,
  value,
  handleValueChange,
  handleRemoveAttRecord,
}) {
  const handleChange = e => {
    handleValueChange(parseInt(e.currentTarget.value, 10));
  };

  return (
    <div className={`row ${styles.attibuteItem}`}>
      <div className={`col-sm-5 ${styles.recordKeyText}`}>
        <FormattedMessage {...messages[keyText]} />
      </div>
      <div className="col-sm-5">
        <input
          className="form-control"
          onChange={handleChange}
          value={value}
          type="number"
        />
      </div>
      <div className="col-sm-2">
        <button
          className="btn btn-light"
          type="button"
          onClick={handleRemoveAttRecord}
        >
          <Trash size={24} />
        </button>
      </div>
    </div>
  );
}

RecordAttribute.propTypes = {
  keyText: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  handleValueChange: PropTypes.func.isRequired,
  handleRemoveAttRecord: PropTypes.func.isRequired,
};

export default RecordAttribute;
