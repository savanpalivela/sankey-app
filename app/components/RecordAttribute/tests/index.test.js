/**
 *
 * Tests for RecordAttribute
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import RecordAttribute from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<RecordAttribute />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <RecordAttribute
          keyText="operations"
          value={100}
          handleValueChange={() => {}}
          handleRemoveAttRecord={() => {}}
        />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });
});
