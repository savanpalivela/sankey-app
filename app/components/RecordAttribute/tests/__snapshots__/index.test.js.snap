// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecordAttribute /> Should render and match the snapshot 1`] = `
<div
  className="row undefined"
>
  <div
    className="col-sm-5 undefined"
  >
    <FormattedMessage
      defaultMessage="Operations"
      id="app.components.RecordAttribute.operations"
      values={Object {}}
    />
  </div>
  <div
    className="col-sm-5"
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="number"
      value={1000}
    />
  </div>
  <div
    className="col-sm-2"
  >
    <button
      className="btn btn-light"
      onClick={[Function]}
      type="button"
    >
      <ForwardRef
        color="currentColor"
        size={24}
      />
    </button>
  </div>
</div>
`;
